### ISSUE TITLE
Race condition: `notifications/tools/list_changed` refresh not awaited before next agent turn causes first tool call to fail

### ISSUE THREAD

**Issue by sauliusvl** [zed-industries/zed#57473]
### Reproduction steps

1. Configure an MCP server whose `tools/list` starts with a small set of meta-tools, one of which dynamically loads additional tools at runtime (similar to a skill-acquisition pattern). 
2. The server advertises `capabilities.tools.listChanged: true` at `initialize`.
3. Call the meta-tool that loads new tools. The server sends `notifications/tools/list_changed` followed by the `tools/call` result on the same SSE stream.
4. The model immediately tries to call one of the newly-loaded tools.

### Current vs. Expected behavior

**Observed:** 
The call fails with an error (tool not found). The model retries and succeeds.

**Expected:** 
The call succeeds on the first attempt.

**Root cause**

In `crates/agent/src/tools/context_server_registry.rs`, the `notifications/tools/list_changed` handler spawns a detached task:

```rust
cx.spawn(async move |cx| {
    this.update(cx, |this, cx| {
        this.reload_tools_for_server(server_id, cx);
    })
}).detach();
```

`reload_tools_for_server` then spawns *another* task that issues the `tools/list` HTTP request and only populates `registered_server.tools` after the response arrives.

Meanwhile, in `Client::handle_input`, the notification is dispatched (handler returns immediately after `.detach()`), and the next loop iteration delivers the `tools/call` response to the awaiting `request()` caller. The agent loop is unblocked and constructs the next LLM inference using the stale tool registry.

The `load_tools: Task<Result<()>>` field on `RegisteredContextServer` already exists but nothing awaits it before the next agent turn.

**Suggested fix**

Before the agent assembles the tool list for the next LLM inference, await any in-flight `load_tools` task for servers that have a pending refresh. This would close the race without changing the MCP wire protocol.

Alternatively, `Client::handle_input` could kick off the `tools/list` request synchronously (registering it as the `load_tools` task) before draining the next message from the channel, so by the time the `tools/call` response resolves, the refresh task is already registered for the agent loop to await.

### Zed version and system specs

Zed: v1.4.1+preview.287.084505b1bbdcf5047ff8438658492ed738a3b9f4 (Zed Preview) 
OS: Linux Wayland arch unknown
Memory: 62.2 GiB
Architecture: x86_64
GPU: Intel(R) Arc(tm) Graphics (MTL) || Intel open-source Mesa driver || Mesa 26.1.1-arch1.2

### Attach Zed log file

<details><summary>Zed.log</summary>

```log

```

</details>

### Relevant Zed settings

<details><summary>settings.json</summary>

```json

```

</details>

### Relevant Keymap

<details><summary>keymap.json</summary>

```json

```

</details>

### If you are using WSL on Windows, what flavor of Linux are you using?

None
